source/common/network/dns_impl.cc

```diff
@@ -28,7 +28,16 @@ DnsResolverImpl::DnsResolverImpl(Event::Dispatcher& dispatcher)
   // launch via main().
   ares_library_init(ARES_LIB_INIT_ALL);
   ares_options options;
+
   initializeChannel(&options, 0);
+
+  if (resolvers.size() > 0) {
+    std::string resolvers_csv = StringUtil::join(resolvers, ",");
+    int result = ares_set_servers_ports_csv(channel_, resolvers_csv.c_str());
+    if (result != ARES_SUCCESS) {
+      throw EnvoyException(fmt::format("dns: error setting resolvers: {}", result));
```

Comment:
This should be a `RELEASE_ASSERT`, since the documentation (https://c-ares.haxx.se/ares_set_servers_csv.html) says the only errors possible in normal use are out-of-memory (see https://github.com/lyft/envoy/blob/master/STYLE.md#error-handling).